crates/node/src/kitchensink.rs

```diff
@@ -0,0 +1,280 @@
+                let _ = child.kill();
+                anyhow::bail!("Timeout waiting for process readiness: {marker}");
+            }
+        }
+    }
+}
+
+impl EthereumNode for KitchensinkNode {
+    fn execute_transaction(
+        &self,
+        transaction: alloy::rpc::types::TransactionRequest,
+    ) -> anyhow::Result<TransactionReceipt> {
+        let url = self.rpc_url.clone();
+        let wallet = self.wallet.clone();
+
+        execute_transaction(Box::pin(async move {
+            Ok(ProviderBuilder::new()
+                .wallet(wallet)
+                .connect(&url)
+                .await?
+                .send_transaction(transaction)
+                .await?
+                .get_receipt()
+                .await?)
+        }))
+    }
+
+    fn trace_transaction(
+        &self,
+        transaction: TransactionReceipt,
+    ) -> anyhow::Result<alloy::rpc::types::trace::geth::GethTrace> {
+        let url = self.rpc_url.clone();
+        let trace_options = GethDebugTracingOptions::prestate_tracer(PreStateConfig {
+            diff_mode: Some(true),
+            disable_code: None,
+            disable_storage: None,
+        });
+
+        let wallet = self.wallet.clone();
+
+        trace_transaction(Box::pin(async move {
+            Ok(ProviderBuilder::new()
+                .wallet(wallet)
+                .connect(&url)
+                .await?
+                .debug_trace_transaction(transaction.transaction_hash, trace_options)
+                .await?)
+        }))
+    }
+}
+
+impl Node for KitchensinkNode {
+    fn new(config: &Arguments) -> Self {
+        let id = NODE_COUNT.fetch_add(1, Ordering::SeqCst);
+        let base_directory = config
+            .directory()
+            .join(Self::BASE_DIRECTORY)
+            .join(id.to_string());
+
+        Self {
+            id,
+            base_directory,
+            substrate_binary: config.kitchensink.clone(),
+            eth_proxy_binary: config.eth_proxy.clone(),
+            rpc_url: String::new(),
+            wallet: config.wallet(),
+            process_substrate: None,
+            process_proxy: None,
+        }
+    }
+
+    fn connection_string(&self) -> String {
+        self.rpc_url.clone()
+    }
+
+    fn shutdown(mut self) -> anyhow::Result<()> {
+        if let Some(mut child) = self.process_proxy.take() {
+            let _ = child.kill();
+        }
+        if let Some(mut child) = self.process_substrate.take() {
+            let _ = child.kill();
+        }
+        Ok(())
+    }
+
+    fn spawn(&mut self, genesis: String) -> anyhow::Result<()> {
+        self.spawn_process(genesis)
+    }
+
+    fn state_diff(&self, transaction: TransactionReceipt) -> anyhow::Result<DiffMode> {
+        match self
+            .trace_transaction(transaction)?
+            .try_into_pre_state_frame()?
+        {
+            PreStateFrame::Diff(diff) => Ok(diff),
+            _ => anyhow::bail!("expected a diff mode trace"),
+        }
+    }
+
```

Comment:
Looking at this now, I believe this should belong to the `NodeInteraction` trait? But thats not your fault and not important for this PR.